### PR TITLE
Closes #88 and #94: Replaced deprecated `admiral` functions in ADSL code

### LINKED ISSUE #88: ADSL Deprecation Warnings after `admiral` Version Update
Getting lifecycle warnings when running ADSL after upgrading to `admiral` v0.10.1, due to derive_vars_disposition_reason() and derive_var_disposition_status() entering the deprecation cycle in `admiral`. 
image

### PR DESCRIPTION
**Pull request by DeclanHodges**
I've replaced calls to `derive_var_disposition_status()` and `derive_vars_disposition_reason()` with `derive_var_merged_cat()` to address `lifecycle` warning messages relating to the future deprecation of the above two `admiral` functions.

### CODE REVIEW

**Comment by DeclanHodges** [comment]
I've also made a small update to remove some new QC mismatches introduced from moving to `admiral` 0.10.1 (see #94) for more details.

File: submission/programs/adsl.R
Comment (laxamanaj) [inline submission/programs/adsl.R:1]: Hi, @DeclanHodges .  Seems @kaz462 also updated these lines 
```
xportr_format(adsl_spec$var_spec %>%
    mutate_at(c("format"), ~ replace_na(., "")), "ADSL") %>%
```
See #91

Comment (DeclanHodges) [reply submission/programs/adsl.R:1]: Thanks @laxamanaj, I'll pull these into the PR.

**Review by laxamanaj — APPROVED** [review]
Thanks for the updates here @DeclanHodges .  After a pull, I ran adsl and all ran fine and QC findings still match up to current QC run. 

 I'm just curious, do you know if this message is warning?

```
`summarise()` has grouped output by 'STUDYID', 'USUBJID'. You can override using the `.groups` argument.
```

**Comment by robertdevine** [comment]
Thanks @laxamanaj and @DeclanHodges, the message appears in the 'dplyr' vignettes (as well).

https://cran.r-project.org/web/packages/dplyr/vignettes/programming.html

**Comment by DeclanHodges** [comment]
@laxamanaj, @robertdevine - after a quick Google, it is a message that refers to `summarise()` dropping the final grouping variable from the list of grouping variables by default. This is not, however, dropping the variable from the output or affecting any results from the code. My understanding is that this would only have any affect if the code didn't specify an ungroup/regroup as necessary for subsequent code requiring grouping. It looks as though this message can be removed or suppressed if needed.

**Pull request merged**